### PR TITLE
Fix issues in `toXml` API and update the docs

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3713 
https://github.com/ballerina-platform/ballerina-standard-library/issues/3716
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -494,7 +494,7 @@ When `attributePrefix` is `&` and `arrayEntryTag` is `list`, the XML representat
 
 ### 4.4. Ballerina record/Map to XML Conversion
 This conversion provides a mapping between the different forms of Ballerina record/Map, to a corresponding matching XML representation.
-The following API returns the XML data to the given Ballerina record/Map. The record has annotations to configure namespaces and attributes, but others don't have these.
+The following API returns the XML data to the given Ballerina record/Map. The record has annotations to configure namespaces and attributes, but other types don't have these.
```
Comment (daneshk) [inline docs/spec/spec.md:497]: can we specify other types here.

File: ballerina/xmldata.bal [hunk 2]
```diff
@@ -47,8 +47,8 @@ public annotation NamespaceConfig Namespace on type;
 public annotation Attribute on record field;
 
 # Converts a `Map` or `Record` representation to its XML representation.
-# The record has annotations to configure namespaces and attributes,
-# but others don't have these.
+# If the data is a defined record type, namespaces and attributes can be
+# defined using annotations. But other types do not have this support.
```
Comment (daneshk) [inline ballerina/xmldata.bal:51]:
```suggestion
# XML `namespaces` and `attributes` can only defined through record annotations are only supported for record values.
```

File: ballerina/xmldata.bal [hunk 3]
```diff
@@ -384,13 +384,14 @@ returns returnType|Error = @java:Method {
 } external;
 
 # Converts an XML to its `Map` or `Record` representation.
-# The namespaces and attributes will not be considered a special case.
+# If `returnType` is not a defined record type, namespaces and attributes are omitted by this conversion.
```
Comment (daneshk) [inline ballerina/xmldata.bal:387]:
```suggestion
# XML `namespaces` and `attributes` can only defined through record annotations are only supported for record values.
```

File: ballerina/xmldata.bal [hunk 4]
```diff
@@ -384,13 +384,14 @@ returns returnType|Error = @java:Method {
...
 #
-# + xmlValue - The XML source to be converted to a given target type
-# + returnType - The `typedesc` of the `map<anydata>` that should be returned as a result
+# + xmlValue - The XML source to be converted to a given target type. If the XML elements have a prefix and
+#              `returnType` is also a defined record, the field names of the record must also have the same prefix.
```
Comment (daneshk) [inline ballerina/xmldata.bal:390]:
```suggestion
# + xmlValue - The XML source to be converted to a given target type. If the XML elements have a prefix, the mapping field names of the record must also have the same prefix.
```

File: ballerina/xmldata.bal [hunk 5]
```diff
@@ -384,13 +384,14 @@ returns returnType|Error = @java:Method {
...
+# + returnType - The `typedesc` of the `Map` or `Record` that should be returned as a result
```
Comment (daneshk) [inline ballerina/xmldata.bal:391]:
```suggestion
# + returnType - The `typedesc` of the returned value. this should be either `map` or `record` type. 
```

File: docs/spec/spec.md (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply docs/spec/spec.md]: Added in 5f5a363

**Review by daneshk — APPROVED** [review]

**Pull request merged**